File: src/main/java/pl/training/module7/breakresponsibilities/after/DeploymentReportFormatter.java
package pl.training.module7.breakresponsibilities.after;

import java.util.Objects;

import pl.training.module7.breakresponsibilities.DeploymentMetrics;

public final class DeploymentReportFormatter {
    public String format(DeploymentMetrics metrics) {
        Objects.requireNonNull(metrics, "metrics");
        return "deployments=" + metrics.deployments()
                + ";failures=" + metrics.failures()
                + ";avgLeadTimeMinutes="
                + metrics.averageLeadTimeMinutes();
    }
}

File: typescript/src/module7/breakresponsibilities/after/DeploymentReportFormatter.ts
import { requireNonNull } from '../../../shared/requireNonNull.js';
import type { DeploymentMetrics } from '../DeploymentMetrics.js';

export class DeploymentReportFormatter {
  format(metrics: DeploymentMetrics): string {
    requireNonNull(metrics, 'metrics');
    return 'deployments=' + metrics.deployments
      + ';failures=' + metrics.failures
      + ';avgLeadTimeMinutes='
      + metrics.averageLeadTimeMinutes;
  }
}
